prisma-fmt/src/references.rs

```diff
@@ -0,0 +1,277 @@
+                    .1
+                    .attributes
+                    .iter()
+                    .find(|attr| extract_ds_from_native_type(attr.name()) == name)
+                    .map(|attr| attr.identifier())
+            })
+            .collect()
+    }
+
+    ctx.db
+        .walk_tops()
+        .flat_map(|top| match top.ast_top() {
+            Top::CompositeType(composite_type) => find_native_type_locations(name, composite_type.iter_fields()),
+            Top::Model(model) => find_native_type_locations(name, model.iter_fields()),
+
+            Top::Enum(_) | Top::Source(_) | Top::Generator(_) => empty_identifiers(),
+        })
+        .collect()
+}
+
+fn find_where_used_as_type<'ast>(ctx: &'ast ReferencesContext<'_>, name: &str) -> Vec<&'ast Identifier> {
+    info!("Get references for top");
+
+    fn get_relevent_identifiers<'a, 'b>(
+        fields: impl Iterator<Item = (FieldId, &'a Field)>,
+        name: &str,
+    ) -> Vec<&'a Identifier> {
+        fields
+            .filter_map(|(_id, field)| match &field.field_type {
+                FieldType::Supported(id) => {
+                    if id.name == name {
+                        Some(id)
+                    } else {
+                        None
+                    }
+                }
+                FieldType::Unsupported(_, _) => None,
+            })
+            .collect()
+    }
+
+    ctx.db
+        .walk_tops()
+        .flat_map(|top| match top.ast_top() {
+            Top::Model(model) => {
+                info!("Get references in model");
+
+                let fields = model.iter_fields();
+
+                get_relevent_identifiers(fields, name)
+            }
+            Top::CompositeType(composite_type) => {
+                info!("Get references in composite type");
+
+                let fields = composite_type.iter_fields();
+                get_relevent_identifiers(fields, name)
+            }
+            Top::Enum(_) | Top::Source(_) | Top::Generator(_) => empty_identifiers(),
+        })
+        .collect()
+}
+
+fn find_where_used_as_top_name<'ast>(ctx: &'ast ReferencesContext<'_>, name: &str) -> Vec<&'ast Identifier> {
+    fn filter_ident<'ast>(id: &'ast Identifier, name: &str) -> Option<&'ast Identifier> {
+        if id.name == name {
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    ctx.db
+        .walk_tops()
+        .filter_map(|top| match top.ast_top() {
+            Top::CompositeType(composite_type) => filter_ident(composite_type.identifier(), name),
+
+            Top::Enum(enm) => filter_ident(enm.identifier(), name),
+
+            Top::Model(model) => filter_ident(model.identifier(), name),
+
+            Top::Source(source) => filter_ident(source.identifier(), name),
+
+            Top::Generator(_) => None,
+        })
+        .collect()
+}
+
+fn extract_ds_from_native_type(attr_name: &str) -> &str {
+    let split = attr_name.split('.').collect::<Vec<&str>>()[0];
+    split
```

Comment:
No need to allocate a new vector

```suggestion
attr_name.split('.').next().unwrap()
```